app/policies/presentation_policy.rb

```diff
@@ -6,4 +6,12 @@ def create?
   def new?
     create?
   end
+
+  def update?
+    (person&.admin? || person&.presentations&.include?(record)).present?
```

Comment:
Sure, that does work. 

I think it would simplify things if you considered a person to not be signed in if they were unpersisted. That way you are always operating on the same object, and not sometimes nil and sometimes a person